src/shared/logger/activation.ts

```diff
@@ -16,13 +15,15 @@ import { recordVscodeViewLogs } from '../telemetry/telemetry'
 import { setLogger } from './logger'
 import { LOG_OUTPUT_CHANNEL } from './outputChannel'
 import { WinstonToolkitLogger } from './winstonToolkitLogger'
+import { ext } from '../extensionGlobals'
 
 const localize = nls.loadMessageBundle()
 
-const LOG_PATH = path.join(getLogBasePath(), 'Code', 'logs', 'aws_toolkit', makeLogFilename())
-const LOG_URI = vscode.Uri.file(path.normalize(LOG_PATH))
 const DEFAULT_LOG_LEVEL: LogLevel = 'info'
 
+/** One log per session */
+let logPath: string
```

Comment:
can we set it here? That better establishes the invariant "is set exactly once".
```suggestion
const logPath = getLogPath()
```

Reply:
Nope, extension globals have not yet loaded yet unfortunately.

Reply:
Yeah. But in theory, activate() could be called by tests or some other accident. Instead, we should guard it somehow...

Reply:
This code is in `getLogPath`:
``
   if (logPath !== undefined) {
        return logPath
    }
``

So it should never changed after being set since `logPath = getLogPath()` only happens in `activate()`